internal/bgp/bgpfrr/config.go

```diff
@@ -0,0 +1,127 @@
+package bgpfrr
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"text/template"
+)
+
+// TODO configFileName will need to be set to the filename of the FRR configuration
+// file in the container.
+var configFileName = "frr.conf"
+
+const configTemplate = `
+log stdout {{.Loglevel}}
+hostname {{.Hostname}}
+
+{{range .Routers -}}
+router bgp {{.Asn}}
+  no bgp default ipv4-unicast
+  bgp router-id {{.RouterId}}
+{{range .Neighbors }}
+  neighbor {{.Addr}} remote-as {{.Asn}}
+{{- end }}
+{{range $n := .Neighbors -}}
+{{range .Advertisements }}
+  address-family {{.Version}} unicast
+    neighbor {{$n.Addr}} activate
+    network {{.Prefix}}
+  exit-address-family
+{{- end}}
+{{end}}
+{{end}}
+`
+
+type frrConfig struct {
+	Loglevel string
+	Hostname string
+	Routers  map[string]*routerConfig
```

Comment:
this just one router/speaker I don't think it needs to be a map

Reply:
no each speak has one myASN and as far as I know we allow only one speaker/node atm

Reply:
But `myASN` is a property of the peer object, not of the speaker, so I _think_ each speaker advertises the same myAsn to the peer, but may have different `myAsn`s if multiple peers are configured. May be wrong here

Reply:
This is a great conversation because it is a little confusing and its good to see I am not the only one confused!

From what I understand, a speaker can peer with multiple peers using different ASNs. i.e. what @fedepaol stated. What I could maybe do differently is use 'myASN' as a key in 'Routers'. I think it is ok to do that as we will not establish two sessions with the same 'myASN' but different 'routerID's? I *think* that is correct?

Reply:
myASN represent '1' this in FRR config
```
router bgp 1
 no bgp default ipv4-unicast
 bgp router-id 10.10.10.10
 neighbor 10.0.0.1 remote-as 64000
 neighbor 10::1 remote-as 20000
 neighbour 10.0.0.1 update-source 20.0.0.2

 address-family ipv4 unicast
  neighbor 10.0.0.1 route-map RMAP in
  neighbor 10.0.0.1 route-map RMAP out
  network 172.18.0.100/24
 exit-address-family

 address-family ipv6 unicast
  neighbor 10::1 activate
  network 1001:1::1/64
 exit-address-family
```

and what I am sayin we can't have today
router bgp 1
router bgp 2
router bgp 3
that means 3 BGP speakers running on one node, remote-as ofcourse you can have as many as you want of them/session it could be same like myASN iBGP session or different eBGP session, same router-id is one/VRF since we are just working with default VRF here then we just need one router-id

Reply:
See previous discussion about this being a requirement discussion.